server/src/main/java/org/candlepin/resource/RoleResource.java

```diff
@@ -30,38 +30,16 @@
 
 import com.google.inject.Inject;
 
-import io.swagger.annotations.Api;
-import io.swagger.annotations.ApiOperation;
-import io.swagger.annotations.ApiParam;
-import io.swagger.annotations.ApiResponse;
-import io.swagger.annotations.ApiResponses;
-import io.swagger.annotations.Authorization;
-
-import org.jboss.resteasy.annotations.providers.jaxb.Wrapped;
 import org.jboss.resteasy.spi.BadRequestException;
 import org.xnap.commons.i18n.I18n;
 
 import java.util.Collection;
-import java.util.stream.Stream;
-
-import javax.ws.rs.Consumes;
-import javax.ws.rs.DELETE;
-import javax.ws.rs.GET;
-import javax.ws.rs.POST;
-import javax.ws.rs.PUT;
-import javax.ws.rs.Path;
-import javax.ws.rs.PathParam;
-import javax.ws.rs.Produces;
-import javax.ws.rs.core.MediaType;
-
-
+import java.util.stream.Collectors;
 
 /**
- *
+ * API implementation for Role operations
  */
-@Path("/roles")
-@Api(value = "roles", authorizations = { @Authorization("basic") })
-public class RoleResource {
+public class RoleResource implements RolesApi {
 
     private UserServiceAdapter userService;
     private OwnerCurator ownerCurator;
```

Comment:
Not part of your changes but ownerCurator and permissionCurator are not used anymore so we could get rid of them.

Reply:
Fixed it.